src/libraries/System.Console/tests/System.Console.Tests.csproj

```diff
@@ -54,9 +53,32 @@
   <ItemGroup Condition="'$(TargetPlatformIdentifier)' != 'windows'">
     <Compile Include="CancelKeyPress.Unix.cs" />
     <Compile Include="NonStandardConfiguration.Unix.cs" />
+    <Compile Include="TermInfo.Unix.cs" />
+    <Compile Include="..\src\System\TermInfo.cs" Link="src\System\TermInfo.cs" />
+    <Compile Include="..\src\System\TermInfo.DatabaseFactory.cs" Link="src\System\TermInfo.DatabaseFactory.cs" />
```

Comment:
FWIW, in other libraries, e.g. all of the System.Net.* libs, System.Text.RegularExpressions, etc., we've split the test projects into FunctionalTests and UnitTests, where the difference is the FunctionalTests are written only in terms of the public surface area exposed by the library under test, and the UnitTests include source from the target library in order to test internals directly.